js/binance.js

```diff
@@ -6053,6 +6054,57 @@ module.exports = class binance extends Exchange {
         };
     }
 
+    async createMarginLoan (code, amount, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        const currency = this.currency (code);
+        const request = {
+            'asset': currency['id'],
+            'amount': amount,
+        };
+        const defaultMargin = this.safeString (params, 'marginMode', 'cross'); // cross or isolated
+        const marginMode = this.safeString2 (this.options, 'defaultMarginMode', 'marginMode', defaultMargin);
+        if (marginMode === 'isolated') {
+            if (symbol === undefined) {
+                throw new BadRequest (this.id + 'createMarginLoan() requires a symbol argument for isolated margin');
+            }
+            request['isIsolated'] = 'TRUE';
+            request['symbol'] = market['id'];
+        }
+        const response = await this.sapiPostMarginLoan (this.extend (request, params));
+        //
+        //     {
+        //         "tranId": 108988250265,
+        //         "clientTag":""
+        //     }
+        //
+        const transaction = this.parseMarginLoan (response, currency['id']);
```

Comment:
We should pass the entire `currency` structure as the second argument to `parseMarginLoan` here for `safeCurrencyCode` to work in `parseMarginLoan`, not just the `currency['id']`.